#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Grass.proto"

WorldInfo {
  basicTimeStep 1
}
Viewpoint {
  orientation -0.16885100344750867 0.09233252964977326 0.9813073130285147 2.1566046761451707
  position 1.6286513388341228 -1.828987017206486 1.1427373467424564
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
  floorAppearance Grass {
  }
}
DEF QuadrupedSimple Robot {
  translation 0 0 0.55
  children [
    Solid {
      children [
        Gyro {
          name "trunk_imu gyro"
        }
        InertialUnit {
          name "trunk_imu inertial"
        }
      ]
      model "imu_link"
      physics Physics {
        density -1
        mass 0.001
        centerOfMass [
          0 0 0
        ]
      }
    }
    Shape {
      appearance DEF base_material PBRAppearance {
        baseColor 0.5 0.5 0.5
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.5 0.24 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 0.146
        anchor 0.28 -0.12 0
      }
      device [
        RotationalMotor {
          name "leg1_hip_joint"
          minPosition -3.14
          maxPosition 3.14
          maxTorque 300
        }
        PositionSensor {
          name "leg1_hip_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.28 -0.12 0
        rotation 1 0 0 0.14600000000000068
        children [
          Pose {
            rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
            children [
              Shape {
                appearance DEF hip_material PBRAppearance {
                  baseColor 0.5 0.5 0.5
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.08
                  radius 0.025
                }
              }
            ]
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              position 0.785
              axis 0 1 0
              anchor 0 -0.08 0
            }
            device [
              RotationalMotor {
                name "leg1_thigh_joint"
                minPosition -3.14
                maxPosition 3.14
                maxTorque 300
              }
              PositionSensor {
                name "leg1_thigh_joint_sensor"
              }
            ]
            endPoint Solid {
              translation 0 -0.08 0
              rotation 0 1 0 0.7850000000000003
              children [
                Pose {
                  translation 0 0 -0.17
                  children [
                    Shape {
                      appearance DEF thigh_material PBRAppearance {
                        baseColor 0.5 0.5 0.5
                        roughness 1
                        metalness 0
                      }
                      geometry Cylinder {
                        height 0.34
                        radius 0.025
                      }
                    }
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    position -1.57
                    axis 0 1 0
                    anchor 0 0 -0.36
                  }
                  device [
                    RotationalMotor {
                      name "leg1_shank_joint"
                      minPosition -3.14
                      maxPosition 3.14
                      maxTorque 300
                    }
                    PositionSensor {
                      name "leg1_shank_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0 -0.36
                    rotation 0 -1 0 1.57
                    children [
                      Pose {
                        translation 0 0 -0.165
                        children [
                          Shape {
                            appearance DEF shank_material PBRAppearance {
                              baseColor 0.5 0.5 0.5
                              roughness 1
                              metalness 0
                            }
                            geometry Cylinder {
                              height 0.33
                              radius 0.015
                            }
                          }
                        ]
                      }
                      Solid {
                        translation 0 0 -0.36
                        children [
                          Shape {
                            appearance DEF toe_material PBRAppearance {
                              baseColor 0.5 0.5 0.5
                              roughness 1
                              metalness 0
                            }
                            geometry Sphere {
                              radius 0.035
                            }
                          }
                        ]
                        name "leg1_toe"
                        boundingObject Sphere {
                          radius 0.035
                        }
                        physics Physics {
                          density -1
                          mass 0.1
                          centerOfMass [
                            0 0 0
                          ]
                          inertiaMatrix [
                            1.6e-05 1.6e-05 1.6e-05
                            0 0 0
                          ]
                        }
                      }
                    ]
                    name "leg1_shank"
                    boundingObject Pose {
                      translation 0 0 -0.165
                      children [
                        Cylinder {
                          height 0.33
                          radius 0.015
                        }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 0.5
                      centerOfMass [
                        0 0 -0.18
                      ]
                      inertiaMatrix [
                        0.0054281 0.0054281 5.625e-05
                        0 0 0
                      ]
                    }
                  }
                }
              ]
              name "leg1_thigh"
              boundingObject Pose {
                translation 0 0 -0.17
                children [
                  Cylinder {
                    height 0.34
                    radius 0.025
                  }
                ]
              }
              physics Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 -0.18
                ]
                inertiaMatrix [
                  0.010956 0.010956 0.0003125
                  0 0 0
                ]
              }
            }
          }
        ]
        name "leg1_hip"
        boundingObject Pose {
          rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
          children [
            Cylinder {
              height 0.08
              radius 0.025
            }
          ]
        }
        physics Physics {
          density -1
          mass 2
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0045792 0.000625 0.0045792
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 0.146
        anchor -0.28 -0.12 0
      }
      device [
        RotationalMotor {
          name "leg2_hip_joint"
          minPosition -3.14
          maxPosition 3.14
          maxTorque 300
        }
        PositionSensor {
          name "leg2_hip_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.28 -0.12 0
        rotation 1 0 0 0.14600000000000068
        children [
          Pose {
            rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
            children [
              Shape {
                appearance USE hip_material
                geometry Cylinder {
                  height 0.08
                  radius 0.025
                }
              }
            ]
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              position 0.785
              axis 0 1 0
              anchor 0 -0.08 0
            }
            device [
              RotationalMotor {
                name "leg2_thigh_joint"
                minPosition -3.14
                maxPosition 3.14
                maxTorque 300
              }
              PositionSensor {
                name "leg2_thigh_joint_sensor"
              }
            ]
            endPoint Solid {
              translation 0 -0.08 0
              rotation 0 1 0 0.7850000000000003
              children [
                Pose {
                  translation 0 0 -0.17
                  children [
                    Shape {
                      appearance USE thigh_material
                      geometry Cylinder {
                        height 0.34
                        radius 0.025
                      }
                    }
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    position -1.57
                    axis 0 1 0
                    anchor 0 0 -0.36
                  }
                  device [
                    RotationalMotor {
                      name "leg2_shank_joint"
                      minPosition -3.14
                      maxPosition 3.14
                      maxTorque 300
                    }
                    PositionSensor {
                      name "leg2_shank_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0 -0.36
                    rotation 0 -1 0 1.57
                    children [
                      Pose {
                        translation 0 0 -0.165
                        children [
                          Shape {
                            appearance USE shank_material
                            geometry Cylinder {
                              height 0.33
                              radius 0.015
                            }
                          }
                        ]
                      }
                      Solid {
                        translation 0 0 -0.36
                        children [
                          Shape {
                            appearance USE toe_material
                            geometry Sphere {
                              radius 0.035
                            }
                          }
                        ]
                        name "leg2_toe"
                        boundingObject Sphere {
                          radius 0.035
                        }
                        physics Physics {
                          density -1
                          mass 0.1
                          centerOfMass [
                            0 0 0
                          ]
                          inertiaMatrix [
                            1.6e-05 1.6e-05 1.6e-05
                            0 0 0
                          ]
                        }
                      }
                    ]
                    name "leg2_shank"
                    boundingObject Pose {
                      translation 0 0 -0.165
                      children [
                        Cylinder {
                          height 0.33
                          radius 0.015
                        }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 0.5
                      centerOfMass [
                        0 0 -0.18
                      ]
                      inertiaMatrix [
                        0.0054281 0.0054281 5.625e-05
                        0 0 0
                      ]
                    }
                  }
                }
              ]
              name "leg2_thigh"
              boundingObject Pose {
                translation 0 0 -0.17
                children [
                  Cylinder {
                    height 0.34
                    radius 0.025
                  }
                ]
              }
              physics Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 -0.18
                ]
                inertiaMatrix [
                  0.010956 0.010956 0.0003125
                  0 0 0
                ]
              }
            }
          }
        ]
        name "leg2_hip"
        boundingObject Pose {
          rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
          children [
            Cylinder {
              height 0.08
              radius 0.025
            }
          ]
        }
        physics Physics {
          density -1
          mass 2
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0045792 0.000625 0.0045792
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -0.146
        anchor 0.28 0.12 0
      }
      device [
        RotationalMotor {
          name "leg3_hip_joint"
          minPosition -3.14
          maxPosition 3.14
          maxTorque 300
        }
        PositionSensor {
          name "leg3_hip_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.28 0.12 0
        rotation -1 0 0 0.14600000000000068
        children [
          Pose {
            rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
            children [
              Shape {
                appearance USE hip_material
                geometry Cylinder {
                  height 0.08
                  radius 0.025
                }
              }
            ]
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              position 0.785
              axis 0 1 0
              anchor 0 0.08 0
            }
            device [
              RotationalMotor {
                name "leg3_thigh_joint"
                minPosition -3.14
                maxPosition 3.14
                maxTorque 300
              }
              PositionSensor {
                name "leg3_thigh_joint_sensor"
              }
            ]
            endPoint Solid {
              translation 0 0.08 0
              rotation 0 1 0 0.7850000000000003
              children [
                Pose {
                  translation 0 0 -0.17
                  children [
                    Shape {
                      appearance USE thigh_material
                      geometry Cylinder {
                        height 0.34
                        radius 0.025
                      }
                    }
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    position -1.57
                    axis 0 1 0
                    anchor 0 0 -0.36
                  }
                  device [
                    RotationalMotor {
                      name "leg3_shank_joint"
                      minPosition -3.14
                      maxPosition 3.14
                      maxTorque 300
                    }
                    PositionSensor {
                      name "leg3_shank_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0 -0.36
                    rotation 0 -1 0 1.57
                    children [
                      Pose {
                        translation 0 0 -0.165
                        children [
                          Shape {
                            appearance USE shank_material
                            geometry Cylinder {
                              height 0.33
                              radius 0.015
                            }
                          }
                        ]
                      }
                      Solid {
                        translation 0 0 -0.36
                        children [
                          Shape {
                            appearance USE toe_material
                            geometry Sphere {
                              radius 0.035
                            }
                          }
                        ]
                        name "leg3_toe"
                        boundingObject Sphere {
                          radius 0.035
                        }
                        physics Physics {
                          density -1
                          mass 0.1
                          centerOfMass [
                            0 0 0
                          ]
                          inertiaMatrix [
                            1.6e-05 1.6e-05 1.6e-05
                            0 0 0
                          ]
                        }
                      }
                    ]
                    name "leg3_shank"
                    boundingObject Pose {
                      translation 0 0 -0.165
                      children [
                        Cylinder {
                          height 0.33
                          radius 0.015
                        }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 0.5
                      centerOfMass [
                        0 0 -0.18
                      ]
                      inertiaMatrix [
                        0.0054281 0.0054281 5.625e-05
                        0 0 0
                      ]
                    }
                  }
                }
              ]
              name "leg3_thigh"
              boundingObject Pose {
                translation 0 0 -0.17
                children [
                  Cylinder {
                    height 0.34
                    radius 0.025
                  }
                ]
              }
              physics Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 -0.18
                ]
                inertiaMatrix [
                  0.010956 0.010956 0.0003125
                  0 0 0
                ]
              }
            }
          }
        ]
        name "leg3_hip"
        boundingObject Pose {
          rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
          children [
            Cylinder {
              height 0.08
              radius 0.025
            }
          ]
        }
        physics Physics {
          density -1
          mass 2
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0045792 0.000625 0.0045792
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -0.146
        anchor -0.28 0.12 0
      }
      device [
        RotationalMotor {
          name "leg4_hip_joint"
          minPosition -3.14
          maxPosition 3.14
          maxTorque 300
        }
        PositionSensor {
          name "leg4_hip_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.28 0.12 0
        rotation -1 0 0 0.14600000000000068
        children [
          Pose {
            rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
            children [
              Shape {
                appearance USE hip_material
                geometry Cylinder {
                  height 0.08
                  radius 0.025
                }
              }
            ]
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              position 0.785
              axis 0 1 0
              anchor 0 0.08 0
            }
            device [
              RotationalMotor {
                name "leg4_thigh_joint"
                minPosition -3.14
                maxPosition 3.14
                maxTorque 300
              }
              PositionSensor {
                name "leg4_thigh_joint_sensor"
              }
            ]
            endPoint Solid {
              translation 0 0.08 0
              rotation 0 1 0 0.7850000000000003
              children [
                Pose {
                  translation 0 0 -0.17
                  children [
                    Shape {
                      appearance USE thigh_material
                      geometry Cylinder {
                        height 0.34
                        radius 0.025
                      }
                    }
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    position -1.57
                    axis 0 1 0
                    anchor 0 0 -0.36
                  }
                  device [
                    RotationalMotor {
                      name "leg4_shank_joint"
                      minPosition -3.14
                      maxPosition 3.14
                      maxTorque 300
                    }
                    PositionSensor {
                      name "leg4_shank_joint_sensor"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0 -0.36
                    rotation 0 -1 0 1.57
                    children [
                      Pose {
                        translation 0 0 -0.165
                        children [
                          Shape {
                            appearance USE shank_material
                            geometry Cylinder {
                              height 0.33
                              radius 0.015
                            }
                          }
                        ]
                      }
                      Solid {
                        translation 0 0 -0.36
                        children [
                          Shape {
                            appearance USE toe_material
                            geometry Sphere {
                              radius 0.035
                            }
                          }
                        ]
                        name "leg4_toe"
                        boundingObject Sphere {
                          radius 0.035
                        }
                        physics Physics {
                          density -1
                          mass 0.1
                          centerOfMass [
                            0 0 0
                          ]
                          inertiaMatrix [
                            1.6e-05 1.6e-05 1.6e-05
                            0 0 0
                          ]
                        }
                      }
                    ]
                    name "leg4_shank"
                    boundingObject Pose {
                      translation 0 0 -0.165
                      children [
                        Cylinder {
                          height 0.33
                          radius 0.015
                        }
                      ]
                    }
                    physics Physics {
                      density -1
                      mass 0.5
                      centerOfMass [
                        0 0 -0.18
                      ]
                      inertiaMatrix [
                        0.0054281 0.0054281 5.625e-05
                        0 0 0
                      ]
                    }
                  }
                }
              ]
              name "leg4_thigh"
              boundingObject Pose {
                translation 0 0 -0.17
                children [
                  Cylinder {
                    height 0.34
                    radius 0.025
                  }
                ]
              }
              physics Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 -0.18
                ]
                inertiaMatrix [
                  0.010956 0.010956 0.0003125
                  0 0 0
                ]
              }
            }
          }
        ]
        name "leg4_hip"
        boundingObject Pose {
          rotation -0.577428933167513 0.5773109331811705 0.5773109331811705 2.09463
          children [
            Cylinder {
              height 0.08
              radius 0.025
            }
          ]
        }
        physics Physics {
          density -1
          mass 2
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0045792 0.000625 0.0045792
            0 0 0
          ]
        }
      }
    }
  ]
  name "QuadrupedSimple"
  boundingObject Box {
    size 0.5 0.24 0.15
  }
  physics Physics {
    density -1
    mass 30
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      0.20025 0.68125 0.769
      0 0 0
    ]
  }
  controller "<extern>"
  supervisor TRUE
}
